californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -321,8 +321,17 @@ public void onTimeout() {
 					}
 
 					@Override
-					public void onResponse(final Response resp) {
-						notificationListener.onNotification(request, resp);
+					public void onResponse(Response response) {
+						if (!response.getOptions().hasObserve()) {
```

Comment:
I would like to see a comment here describing the circumstances under which this case will actually happen. I bet, if we do not document it here, we will all wonder why added this in two months time ;-)